shell/platform/tizen/channels/platform_channel_tizen.cc

```diff
@@ -84,6 +86,87 @@ class FeedbackManager {
   bool initialized_ = false;
 };
 
+#ifdef COMMON_PROFILE
+class TizenWindowSystemShell {
+ public:
+  static TizenWindowSystemShell& GetInstance() {
+    static TizenWindowSystemShell instance;
+    return instance;
+  }
+
+  TizenWindowSystemShell(const TizenWindowSystemShell&) = delete;
+  TizenWindowSystemShell& operator=(const TizenWindowSystemShell&) = delete;
+
+  bool IsSoftkeyShown() { return is_softkey_shown_; }
+
+  void InitializeSoftkey(uint32_t window_id) {
+    if (tizen_shell_softkey_ || !tizen_shell_) {
+      return;
+    }
+    tizen_shell_softkey_ = tzsh_softkey_create(tizen_shell_, window_id);
+    if (!tizen_shell_softkey_) {
+      int ret = get_last_result();
+      if (ret == TZSH_ERROR_PERMISSION_DENIED) {
+        FT_LOG(Error) << "Permission denied. You need a "
+                         "\"http://tizen.org/privilege/windowsystem.admin\" "
+                         "privilege to use this method.";
+      } else {
+        FT_LOG(Error) << "tzsh_softkey_create() failed with error: "
+                      << get_error_message(ret);
+      }
+    }
+  }
+
+  void ShowSoftkey() {
+    if (!tizen_shell_softkey_) {
+      return;
+    }
+    int ret = tzsh_softkey_global_show(tizen_shell_softkey_);
+    if (ret != TZSH_ERROR_NONE) {
+      FT_LOG(Error) << "tzsh_softkey_global_show() failed with error: "
+                    << get_error_message(ret);
+    }
+    is_softkey_shown_ = true;
+  }
+
+  void HideSoftkey() {
+    if (!tizen_shell_softkey_) {
+      return;
+    }
+    // Always show the softkey before hiding it again, to avoid subtle bugs.
+    tzsh_softkey_global_show(tizen_shell_softkey_);
+    int ret = tzsh_softkey_global_hide(tizen_shell_softkey_);
+    if (ret != TZSH_ERROR_NONE) {
+      FT_LOG(Error) << "tzsh_softkey_global_hide() failed with error: "
+                    << get_error_message(ret);
+    }
+    is_softkey_shown_ = false;
```

Comment:
Ditto.

Reply:
> Ditto.

A word that reminds me someone.. (Never mind!, it's a joke for @bbrto21 :smile: )